drivers/display/display_mcux_dcnano_lcdif.c

```diff
@@ -0,0 +1,319 @@
+	uint32_t h_idx;
+	const uint8_t *src;
+	uint8_t *dst;
+
+	__ASSERT((data->pixel_bytes * desc->pitch * desc->height) <=
+		desc->buf_size, "Input buffer too small");
+
+	LOG_DBG("W=%d, H=%d @%d,%d", desc->width, desc->height, x, y);
+
+#ifdef CONFIG_MCUX_DCNANO_LCDIF_DOUBLE_FRAMEBUFFER
+	/* If display write is partial, copy current framebuffer to
+	 * queued one. Note- this has a significant performance
+	 * impact, especially when using external RAM.
+	 */
+	if ((x != 0) ||
+		(y != 0) ||
+		(desc->height != config->dpi_config.panelHeight) ||
+		(desc->width != config->dpi_config.panelWidth)) {
+		memcpy(data->fb[next_fb_idx], data->fb[data->fb_idx],
+			config->fb_bytes);
+	}
+#endif
+
+
+	src = buf;
+	dst = data->fb[next_fb_idx];
+	dst += data->pixel_bytes * ((y * config->dpi_config.panelWidth) + x);
+
+	for (h_idx = 0; h_idx < desc->height; h_idx++) {
+		memcpy(dst, src, data->pixel_bytes * desc->width);
+		src += data->pixel_bytes * desc->pitch;
+		dst += data->pixel_bytes * config->dpi_config.panelWidth;
+	}
+
+#if defined(CONFIG_HAS_MCUX_CACHE) && defined(CONFIG_MCUX_DCNANO_LCDIF_MAINTAIN_CACHE)
+	CACHE64_InvalidateCacheByRange((uint32_t) data->fb[next_fb_idx],
+					config->fb_bytes);
+#endif
+
+
+	/* Wait for framebuffer completion before writing */
+	k_sem_take(&data->sem, K_FOREVER);
+
+	/* Set new framebuffer */
+	LCDIF_SetFrameBufferStride(config->base, 0,
+		config->dpi_config.panelWidth * data->pixel_bytes);
+	LCDIF_SetFrameBufferAddr(config->base, 0,
+		(uint32_t)data->fb[next_fb_idx]);
+	LCDIF_SetFrameBufferConfig(config->base, 0, &data->fb_config);
+
+	/* Update current framebuffer IDX */
+	data->fb_idx = next_fb_idx;
+
+	return 0;
+}
+
+static void mcux_dcnano_lcdif_get_capabilities(const struct device *dev,
+		struct display_capabilities *capabilities)
+{
+	const struct mcux_dcnano_lcdif_config *config = dev->config;
+	struct mcux_dcnano_lcdif_data *data = dev->data;
+
+	capabilities->y_resolution = config->dpi_config.panelHeight;
+	capabilities->x_resolution = config->dpi_config.panelWidth;
+	capabilities->supported_pixel_formats =
+		(PIXEL_FORMAT_BGR_565 | PIXEL_FORMAT_ARGB_8888);
+	capabilities->current_orientation = DISPLAY_ORIENTATION_NORMAL;
+	switch (data->fb_config.format) {
+	case kLCDIF_PixelFormatRGB565:
+		/* Zephyr stores RGB565 as big endian, and LCDIF
+		 * expects little endian. Use BGR565 format to resolve
+		 * this.
+		 */
+		capabilities->current_pixel_format = PIXEL_FORMAT_BGR_565;
+		break;
+	case kLCDIF_PixelFormatXRGB8888:
+		capabilities->current_pixel_format = PIXEL_FORMAT_ARGB_8888;
+		break;
+	default:
+		/* Other LCDIF formats don't have a Zephyr enum yet */
+		return;
+	}
+}
+
+static void *mcux_dcnano_lcdif_get_framebuffer(const struct device *dev)
+{
+#ifdef CONFIG_MCUX_DCNANO_LCDIF_DOUBLE_FRAMEBUFFER
+	return NULL; /* Direct framebuffer access not supported */
+#else
+	const struct mcux_dcnano_lcdif_config *config = dev->config;
+
+	return config->fb_ptr;
+#endif
```

Comment:
Seems like IS_ENABLED(CONFIG_MCUX_DCNANO_LCDIF_DOUBLE_FRAMEBUFFER) can be used here.
```c
	const struct mcux_dcnano_lcdif_config *config = dev->config;

	if (IS_ENABLED(CONFIG_MCUX_DCNANO_LCDIF_DOUBLE_FRAMEBUFFER)) {
		return config->fb_ptr;
	} else {
		return NULL;
	}

```

Reply:
Sure, I've changed this function to use `IS_ENABLED`